### DISCUSSION TITLE
[Error with latest PackerUpdate] lspsaga.nvim v0.2.3+ has breaking change. Please read the docs ...

### DISCUSSION THREAD

**Discussion by sananand007** [nvimdev/lspsaga.nvim#714] [Q&A] [answered]
The error that I see after the `PackerUpdate` 

```
Error detected while processing /Users/<username>/.config/nvim/plugin/lspsaga.rc.lua:
lspsaga.nvim v0.2.3+ has breaking changes. Please read the docs and migrate your configuration to the new "setup" function!
```

could someone please help me understand What changed and how can I fix it?

My `lspsaga.rc.lua` is as below 

```
local status, saga = pcall(require, "lspsaga")
if (not status) then return end

saga.init_lsp_saga {
  server_filetype_map = {
    typescript = 'typescript'
  }
}

local opts = { noremap = true, silent = true }
local keymap = vim.keymap.set
keymap('n', '<C-j>', '<Cmd>Lspsaga diagnostic_jump_next<CR>', opts)
keymap('n', 'K', '<Cmd>Lspsaga hover_doc<CR>', opts)
keymap('n', 'gd', '<Cmd>Lspsaga lsp_finder<CR>', opts)
-- vim.keymap.set('i', '<C-k>', '<Cmd>Lspsaga signature_help<CR>', opts)
keymap('i', '<C-k>', '<Cmd>Lspsaga signature_help<CR>', opts)
keymap('n', 'gp', '<Cmd>Lspsaga peek_definition<CR>', opts)
keymap('n', 'gr', '<Cmd>Lspsaga rename<CR>', opts)
```

**Comment by glepnir** [comment]
did you read this message .read docs and new setup function. 

`lspsaga.nvim v0.2.3+ has breaking changes. Please read the docs and migrate your configuration to the new "setup" function!`

**Reply by sananand007** [reply]
thanks for the reply @glepnir . Yes I did read that, but that is not telling me where the error is and what I need to modify ( not a lua expert here) 

I did read the Docs, but I am not clear which is the `new setup` that I need to pick up ?

**Comment by sananand007** [comment, marked as answer]
✅ https://github.com/glepnir/lspsaga.nvim/commit/afbd5718ffd453662ada49e9c520df60efc8f444#diff-40f5296ccddd71714533f9f84f3dc94738621a2c4b330f6a3334c988b918f59cR126

I got it.. tnx again !

**Comment by arnoldmartinez** [comment]
me too I have the same problema when I migrated to another computer and the plugins was upgraded.
